### PR TITLE
Add non-inplace operator overloads to :class:`~.ValueTracker`

### PR DESCRIPTION
**Pull request by kilacoda**
## Changelog / Overview

Added methods `__add__`, `__sub__`, `__floordiv__`, `__mul__`, `__truediv__`, `__mod__`, `__pow__`, `__radd__`, `__rmul__` to :class:`~.ValueTracker`

## Explanation for Changes

Adds operator overrides for non-inplace operations to provide more flexibility in the use of ValueTrackers.

## Documentation Reference
 
https://docs.manim.community/en/stable/reference/manim.mobject.value_tracker.ValueTracker.html

## Testing Status

Tested with `pytest test_value_tracker.py`, 23/23 tests pass
## Further Comments

N/A
## Checklist
- [x] I have read the [Contributing Guidelines](https://docs.manim.community/en/latest/contributing.html)
- [x] I have written a descriptive PR title (see top of PR template for examples)
- [ ] I have written a changelog entry for the PR or deem it unnecessary
- [x] My new functions/classes either have a docstring or are private
- [x] My new functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] My new documentation builds, looks correctly formatted, and adds no additional build warnings

## Reviewer Checklist
- [ ] The PR title is descriptive enough
- [ ] The PR is labeled correctly
- [ ] The changelog entry is completed if necessary
- [ ] Newly added functions/classes either have a docstring or are private
- [ ] Newly added functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] Newly added documentation builds, looks correctly formatted, and adds no additional build warnings

### CODE REVIEW

**Comment by PhotonSpheres** [comment]
Just curios, why the change in the of Sphinx `docs/requirements.txt` here? Doesn't seem to be related to the PR, no?

**Comment by kilacoda** [comment]
It wasn't my doing, must have come along when I was updating the branch.

**Comment by Darylgolden** [comment]
I updated the branch from GH and they're successfully removed.

File: manim/mobject/value_tracker.py [hunk 1]
```diff
@@ -92,41 +92,83 @@ def __bool__(self):
         """Return whether the value of this value tracker evaluates as true."""
         return bool(self.get_value())
 
+    def __add__(self, value: Union[int, float]):
```
Comment (Darylgolden) [inline manim/mobject/value_tracker.py:95]:
```suggestion
    def __add__(self, value: float):
```

Comment (Darylgolden) [reply manim/mobject/value_tracker.py:95]: We've previously discussed and agreed that just using `float` as the typehint is fine.

File: manim/mobject/value_tracker.py [hunk 2]
```diff
@@ -92,41 +92,83 @@ def __bool__(self):
...
+        """
+        self.increment_value(value)
+        return self
+
     def __iadd__(self, d_value: float):
         """adds ``+=`` syntax to increment the value of the ValueTracker"""
         self.increment_value(d_value)
         return self
 
+    def __floordiv__(self, d_value: Union[int, float]):
```
Comment (Darylgolden) [inline manim/mobject/value_tracker.py:108]:
```suggestion
    def __floordiv__(self, d_value: float):
```

**Review by Darylgolden — CHANGES_REQUESTED** [review]
Minor typehint changes, otherwise LGTM!

**Comment by hydrobeam** [comment]
If `__add__`  (and the rest) is defined and the implementations for `__add__` and `__iadd__` are the same, wouldn't it be more convenient to just drop the `__iadd__`'s since it falls back to `__add__`? From the python docs:

> For instance, if x is an instance of a class with an `__iadd__()` method, `x += y` is equivalent to `x = x.__iadd__(y)` . Otherwise, `x.__add__(y)` and `y.__radd__(x)` are considered,

https://docs.python.org/3/reference/datamodel.html#emulating-numeric-types

**Comment by Darylgolden** [comment]
Hi, are you still planning on working on this?

**Comment by behackl** [comment]
Closing this due to inactivity. Please re-open in case you want to continue working on this. The docstrings currently don't look right to me.

**Pull request closed without merging**